### DISCUSSION TITLE
RG: Ignoring dirs

### DISCUSSION THREAD

**Discussion by barrettruth** [ibhagwan/fzf-lua#1289] [General]
how the heck do i ignore a directory with `rg_opts` and fzf-lua

it works perfectly via cli but not when I pass identical args to rg_opts

**Comment by ibhagwan** [comment]
> it works perfectly via cli but not when I pass identical args to rg_opts

Whatever works in the cli should work in fzf-lua, run `:FzfLua live_grep debug=true` and explore the underlying command as you type, fzf-lua merely displays the output of the shell command.

Alternatively you can also use `file_ignore_patterns` which uses lua pattern matching as part of the output processing to ignore certain patterns, but IMHO using `cmd|rg_opts` is preferable as it would be filtered at the shell leading to better performance.

**Comment by barrettruth** [comment]
My config mistake.

BTW, for anyone else reading this, I've made my fzf-lua configs identical both in the shell and neovim like so:

`.zprofile`:
```zsh
export FZF_LUA_RG_OPTS="$(tr '\n' ' ' < "$RIPGREP_CONFIG_PATH")"
```

fzf setup:
```lua
opts = {
        files = {
            fd_opts = vim.env.FZF_CTRL_T_COMMAND:match(' (.*)'),
        },
        fzf_args = vim.env.FZF_DEFAULT_OPTS,
        grep = {
            rg_opts = vim.env.FZF_LUA_RG_OPTS,
        }
}
```

**Reply by ibhagwan** [reply]
> RIPGREP_CONFIG_PATH

If you want this considered in fzf-lua I can easily add an option that you can pass under grep which will be relayed to the main process:
```lua
        grep = {
            RIPGREP_CONFIG_PATH = vim.env.RIPGREP_CONFIG_PATH,
        }
```

Lmk if you find this valuable I can add it pretty quick.

**Reply by ibhagwan** [reply]
See https://github.com/ibhagwan/fzf-lua/commit/f5240e364df3d3b954205aecf0c860cd7b185c06, use:
```lua
require("fzf-lua").setup({
  grep = { RIPGREP_CONFIG_PATH = vim.env.RIPGREP_CONFIG_PATH }
})
```

**Comment by barrettruth** [comment]
@ibhagwan hate to say this, but [there is probably never going to be a global fd config file](https://github.com/sharkdp/fd/issues/362). - I recommend removing `FD_CONFIG_PATH` for the sake of correctness - it is also completely meaningless to fd as of now.

**Reply by barrettruth** [reply]
I also think allowing someone to manually specify `RIPGREP_CONFIG_PATH` is an intelligent compromise with respect to your previous conversation with folke.

Thanks for the quick response as always.

**Reply by ibhagwan** [reply]
> `FD_CONFIG_PATH`

You put it in my head in #1288 lol
> Similarly with FD and `~/.config/fd/config.toml` with `export FD_CONFIG_PATH=~.config/fd/config.toml`

> I recommend removing FD_CONFIG_PATH for the sake of correctness - it is also completely meaningless to fd as of now.

It’s somewhat harmless but I’ll remove it nonetheless.

**Reply by ibhagwan** [reply]
https://github.com/ibhagwan/fzf-lua/commit/f32e91c2d4f645e746d8bef97346e24f6841a849

**Reply by barrettruth** [reply]
> > `FD_CONFIG_PATH`
> 
> You put it in my head in #1288 lol
> 
> > Similarly with FD and `~/.config/fd/config.toml` with `export FD_CONFIG_PATH=~.config/fd/config.toml`
> 
> > I recommend removing FD_CONFIG_PATH for the sake of correctness - it is also completely meaningless to fd as of now.
> 
> It’s somewhat harmless but I’ll remove it nonetheless.

LOL that was me believing GPT fabrication - sorry about that. Trying to get in the habit of doing more research before asking questions.

**Reply by ibhagwan** [reply]
Btw you don’t need to use `fd_opts` for files, for what you need why not use `cmd` directly (eliminating the `match` call)?
```lua
        files = {
            cmd = vim.env.FZF_CTRL_T_COMMAND,
        },
```

**Reply by barrettruth** [reply]
ITS SO CLEAN









ty

**Reply by ibhagwan** [reply]
While we’re at it, fix the typo you have in quick fix :)
https://github.com/barrett-ruth/dots/blob/dc9cbca3ce57152b0e0cbaa3e9cf8fb7424fc91c/nvim/lua/plugins/fzf.lua#L81

Also, if you haven’t seen yet it’s worth exploring the new formatters:
```lua
formatter = "path.dirname_first"  -- directory has a special highlight group FzfLuaDirPart
formatter = "path.filename_first" -- VScode-like format where filename is first
formatter = { "path.filename_first", v = 2 } -- same as above with hidden path part for logical search (so you can still search /foo/bar

**Reply by barrettruth** [reply]
No way, this is super cool. fzf clears telescope in my humble ~~objective~~ subjective opinion

**Reply by barrettruth** [reply]
Hey sorry, two last things:

1. [typo](https://github.com/ibhagwan/fzf-lua/blob/f32e91c2d4f645e746d8bef97346e24f6841a849/README.md?plain=1#L984)

2. I pretty much only use `live_grep_native`, and passing the `formatter` config (string or table) to my opts doesn't work:
```lua
        grep = {
            RIPGREP_CONFIG_PATH = vim.env.RIPGREP_CONFIG_PATH,
            no_header_i = true,
            formatter = { 'path.filename_first' }
        },
```
while the following works:
```lua
FzfLua live_grep_native formatter=path.filename_first
```

I've noticed all the other settings (two right now) I pass to the grep key get "inherited" by all fzf-lua grep-related functions - except for formatter. Is there a discrepancy or am I missing something here?

**Reply by ibhagwan** [reply]
> I pretty much only use live_grep_native, and passing the formatter config (string or table) to my opts doesn't work

This is by design, the "native" version is designed for max performance and, as its name suggests, is native to the shell i.e. the `rg` command is piped directly to fzf, this prevents any processing from being done by the lua wrapper code (as it's never called).

Any type of lua processing `path_shorten`, `file_ignore_patterns`, `file_icons`, `git_icons`, formatters, etc, will never work with `live_grep_native`, this also applies to glob processing so you can never do `foo -- *.lua` to search only lua files, which you can do with `live_grep_glob` (or if you set `grep = { rg_glob=true }`).

**Reply by ibhagwan** [reply]
> typo

https://github.com/ibhagwan/fzf-lua/commit/af67bd215614f387544f323ef2e5f643af79c842

Ty
